### PR TITLE
feature: S3C-2273 Add unit and functional testing

### PR DESCRIPTION
**Pull request by bennettbuchanan**
Follow up test cases to cover `UtapiReindex` functionality thoroughly.

### CODE REVIEW

**Review by anurag4DSB — APPROVED** [review]

File: tests/functional/testReindex.js [hunk 1]
```diff
@@ -37,33 +100,173 @@ describe('UtapiReindex', () => {
...
+            });
+        });
+
+        describe('lock is already acquired', () => {
+            beforeEach(done => {
+                shouldAcquireLock(done);
+            });
+
+            it('should not acquire the lock key', done => {
+                shouldReleaseLock(done);
```
Comment (jonathan-gramain) [inline tests/functional/testReindex.js:135]: did you mean to call `shouldAcquireLock()` here?

File: tests/functional/testReindex.js [hunk 2]
```diff
@@ -37,33 +100,173 @@ describe('UtapiReindex', () => {
...
+            it('should release the lock key', done => {
+                reindex._attemptUnlock(); // Lock should be released here.
+                setTimeout(() => shouldNotReleaseLock(done), 200);
+            });
+        });
+    });
+
+    describe('::_scheduleJob', () => {
+        function waitUntilLockHasValue({ value }, cb) {
+            let shouldWait;
```
Comment (jonathan-gramain) [inline tests/functional/testReindex.js]: It seems the meaning of this variable is reversed as it is used, you probably meant `shouldLeave`

File: tests/functional/testReindex.js [hunk 3]
```diff
@@ -37,33 +100,173 @@ describe('UtapiReindex', () => {
...
+                const { storageUtilized, numberOfObjects } = expected;
+                assert.deepStrictEqual(res[0].storageUtilized, storageUtilized);
+                assert.deepStrictEqual(res[0].numberOfObjects, numberOfObjects);
+                return cb();
+            });
+        }
+
+        beforeEach(done => {
+            reindex._scheduleJob();
+            // Wait until the scripts have started and finsihed reindexing.
```
Comment (jonathan-gramain) [inline tests/functional/testReindex.js]: typo in "finisihed"

File: tests/functional/testReindex.js (hunk 2 above)
Comment (bennettbuchanan) [reply tests/functional/testReindex.js]: Thanks! I had changed `async.doWhile` to `async.doUntil` and neglected the naming of this variable.

File: tests/functional/testReindex.js (hunk 1 above)
Comment (bennettbuchanan) [reply tests/functional/testReindex.js:135]: Looks like the test description is incorrect here. The test case is correct (i.e. using `shouldReleaseLock`, but the description should be more like "should release the lock key". Thanks for that catch.

File: tests/functional/testReindex.js (hunk 2 above)
Comment (bennettbuchanan) [reply tests/functional/testReindex.js]: Updated.

**Review by jonathan-gramain — APPROVED** [review]

**Comment by bennettbuchanan** [comment]
@bert-e reset

**Comment by bennettbuchanan** [comment]
@bert-e reset

**Comment by bennettbuchanan** [comment]
@bert-e approve

**Pull request merged**